source/Calamari/Variables/DeploymentJournalVariableContributor.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Linq;
+using Calamari.Deployment;
+using Calamari.Deployment.Journal;
+using Calamari.Integration.FileSystem;
+using Calamari.Integration.Processes.Semaphores;
+
+namespace Calamari.Variables
+{
+    static class DeploymentJournalVariableContributor
+    {
+        public static void Contribute(ICalamariFileSystem fileSystem, IVariables variables)
+        {
+            var policySet = variables.Get(SpecialVariables.RetentionPolicySet);
+            if (string.IsNullOrWhiteSpace(policySet))
+                return;
+
+            var journal = new DeploymentJournal(fileSystem, SemaphoreFactory.Get(), variables);
+            Previous(variables, journal, policySet);
+            PreviousSuccessful(variables, journal, policySet);
+        }
+
+        internal static void Previous(IVariables variables, IDeploymentJournal journal, string policySet)
+        {
+            var previous = journal.GetLatestInstallation(policySet);
+            
+            if (previous == null)
+            {
+                variables.Set(SpecialVariables.Tentacle.PreviousInstallation.OriginalInstalledPath, "");
+                variables.Set(SpecialVariables.Tentacle.PreviousInstallation.CustomInstallationDirectory, "");
+                variables.Set(SpecialVariables.Tentacle.PreviousInstallation.PackageFilePath, "");
+                variables.Set(SpecialVariables.Tentacle.PreviousInstallation.PackageVersion, "");
+            }
+            else
+            {
+                // This is assuming this convention is used only in steps with only one package
```

Comment:
Is this comment still valid? It's not a convention anymore, but is the single package still an assumption? I'd assume yes, but just checking